lms/djangoapps/courseware/views/index.py

```diff
@@ -132,6 +133,8 @@ def get(self, request, course_id, chapter=None, section=None, position=None):
 
                 self.is_staff = has_access(request.user, 'staff', self.course)
                 self._setup_masquerade_for_effective_user()
+                register_course_expired_message(request, self.course)
```

Comment:
shouldn't this be behind waffle flag?

Reply:
actually nm, just you check the flag in the function

Reply:
right now this method is called in both places but should do nothing because the flag is checked on the first line. adding the flag check in one place seemed a bit simpler, but i'm open to doing it differently if you have very strong opinions